### PR TITLE
Move WPF to separate nuspec

### PR DESCRIPTION
**Pull request by pauldendulk**
See discussion here:
https://github.com/Mapsui/Mapsui/issues/1207

### CODE REVIEW

File: NuSpec/Mapsui.Wpf.nuspec [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    </description>
+    <tags>map maps mapping geo gis osm</tags>
+    <copyright>© Mapsui Contributers</copyright>
+    <language>en-US</language>
+    <dependencies>
+      <!-- [ or ] is including, ( or ) is excluding -->
+      <group targetFramework="net48">
+        <dependency id="Mapsui" version="$version$"/>
+        <dependency id="BruTile" version="[4.0.0,5.0.0)"/>
+        <dependency id="Newtonsoft.json" version="[11.0.1,)"/>
```
Comment (janusw) [inline NuSpec/Mapsui.Wpf.nuspec:22]: I think the dependencies need to be cleaned up here: Newtonsoft.Json, RichTextKit and Svg.Skia do not seem to be required (at least).

File: NuSpec/Mapsui.Wpf.nuspec [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  <metadata>
+    <id>Mapsui.Wpf</id>
+    <version>$version$</version>
+    <title>Mapsui</title>
+    <authors>Paul den Dulk</authors>
+    <owners>Paul den Dulk</owners>
+    <projectUrl>https://github.com/Mapsui/Mapsui</projectUrl>
+    <requireLicenseAcceptance>false</requireLicenseAcceptance>
+    <description>
+      Mapsui - Library for mapping (native UI for different platforms)
```
Comment (janusw) [inline NuSpec/Mapsui.Wpf.nuspec:12]: The description should be adjusted ("UI for WPF" or similar) ...

**Review by janusw — CHANGES_REQUESTED** [review]

**Comment by janusw** [comment]
In general I'm okay with this, but I think we should go further with the separation (only splitting off WPF but leaving the rest together seems inconsistent).

If you're in a hurry with this, Paul, it can be merged when the minor comments above are fixed. But I would vote for continuing to separate packages in a follow-up PR.

We probably do not want to change the packaging scheme for every minor release, so we should really agree on a scheme, implement it and then make a final release 3.0.

**Comment by janusw** [comment]
And one more comment: It's a bit of a shame that this sort of problem is not caught by any of the automatic CI builds.

Can we improve this situation somehow (without too much effort)?

**Comment by pauldendulk** [comment]
> And one more comment: It's a bit of a shame that this sort of problem is not caught by any of the automatic CI builds.
> Can we improve this situation somehow (without too much effort)?

This is an advanced test. I don't have this in my other projects. I assume others do something like this, perhaps we can learn from their workflow. There might be tools for it. 

What we could do is create a script that updates some code to the [latest nuget](https://mapsui.com/documentation/nuget-of-latest-build.html?q=nuget) and runs some tests. It will be easier to test the core Mapsui project. For UI projects it will be harder. We could build the 'getting started' projects and at least test if it builds. It would catch the problem we just saw.

**Comment by janusw** [comment]
> What we could do is create a script that updates some code to the [latest nuget](https://mapsui.com/documentation/nuget-of-latest-build.html?q=nuget) and runs some tests. It will be easier to test the core Mapsui project. For UI projects it will be harder. We could build the 'getting started' projects and at least test if it builds. It would catch the problem we just saw.

Well, I don't think we need any advanced UI tests for the start. Simply having the 'getting started' guides as VS projects and building them (without project references to Mapsui core, but using the latest nupkgs) should be enough. (Also there is already some value in having getting-started projects, instead of just describing how to set them up.)

Would be nice to do that sometime, but it should not be part of this PR. Please just address my review comments above, then this can be merged. Or do you want me to take over? This is the most urgent PR we have right now.

**Comment by pauldendulk** [comment]
> Would be nice to do that sometime, but it should not be part of this PR. Please just address my review comments above, then this can be merged. Or do you want me to take over? This is the most urgent PR we have right now.

The thing is you made me think about the alternatives. Perhaps we should jump to a solution with [nugetizer](https://www.nuget.org/packages/nugetizer). I will look at this the coming days.

**Comment by janusw** [comment]
> > Would be nice to do that sometime, but it should not be part of this PR. Please just address my review comments above, then this can be merged. Or do you want me to take over? This is the most urgent PR we have right now.
> 
> The thing is you made me think about the alternatives. Perhaps we should jump to a solution with [nugetizer]

I don't really know nugetizer and I only had a quick look at it, but from what I understand it seems like this mightly only help us get rid of the nuspec files (not sure if its worth to go there). In any case, we still have to decide how to cut our packages.

If you're not planning to go ahead with this PR in a timely manner, I guess we should at least fix Mapsui.Native.nuspec, because right now it is simply broken.

**Comment by janusw** [comment]
> If you're not planning to go ahead with this PR in a timely manner, I guess we should at least fix Mapsui.Native.nuspec, because right now it is simply broken.

See #1214.

File: NuSpec/Mapsui.Wpf.nuspec (hunk 2 above)
Comment (pauldendulk) [reply NuSpec/Mapsui.Wpf.nuspec]: Updated https://github.com/Mapsui/Mapsui/pull/1208/commits/bffc01ef07db36432231a8a937c8a7d3f240dbdb

File: NuSpec/Mapsui.Wpf.nuspec (hunk 1 above)
Comment (pauldendulk) [reply NuSpec/Mapsui.Wpf.nuspec]: Updated https://github.com/Mapsui/Mapsui/pull/1208/commits/eabeab3a1cd444459457106828fc96ef53faa659

**Comment by pauldendulk** [comment]
@janusw I processed your remark. You can review again.

**Review by janusw — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**